spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraTemplate.java

```diff
@@ -723,16 +762,21 @@ private int getEffectiveFetchSize(Statement statement) {
 	}
 
 	@SuppressWarnings("unchecked")
-	private <T> Function<Row, T> getMapper(Class<?> entityType, Class<T> targetType) {
+	private <T> Function<Row, T> getMapper(Class<?> entityType, Class<T> targetType, boolean emitEvents) {
```

Comment:
It would make sense to add the table name as an additional argument instead resolving it from the entity type. Is there any case we would not want to emit an event?

Reply:
Currently there is no use-case for not emitting events, but I felt that this should be configurable. In the end, caller would like to receive a mapper and may not care about triggering events. Matter of taste.

Reply:
We can add that kind of things always at a later stage. This method is `private` and nothing breaks if we change that method at a later time.